### DISCUSSION TITLE
It is possible jk-bms and atorch-dl24 on the same ESP32?

### DISCUSSION THREAD

**Discussion by linuxdep** [syssi/esphome-atorch-dl24#38] [General]
Hi, nice project, many thanks for your work!

It is possible both jk-bms and dl24 at same ESP? 
I'm still considering whether it really makes sense, since the dl24 is not always active, but then saves an ESP. This would have always saved the data from the DL24 in HA.
Has anyone ever done it this way?

**Comment by syssi** [comment]
Should be possible! If I remember correctly a ESP32 supports up to 3 BLE client connections because of resource limits (please check the info box at the official `ble_client` documentation). You could add the `switch` of the `ble_client` component to disable the BLE client connection if the DL24 isn't in use.

**Reply by linuxdep** [reply]
Thanks, my CAM modul don't work with esp-idf... Since the CAM is already on the board anyway, it would have been great to have an eye on the battery as well... but not that important. Then I try to get the DL24 and BMS to run on one device. Then I can at least take part in the minutes of tests. Thanks again for your great work on the projects.

**Reply by syssi** [reply]
Hmm.. running two BLE clients and processing a camera sensor is a bit much. ;-)
